packages/core/admin/admin/src/core/fetch-client/index.js

```diff
@@ -0,0 +1,38 @@
+import axios from 'axios';
+import { auth } from '@strapi/helper-plugin';
+
+const instance = axios.create({
+  baseURL: process.env.STRAPI_ADMIN_BACKEND_URL,
+});
+
+instance.interceptors.request.use(
+  async (config) => {
+    config.headers = {
+      Authorization: `Bearer ${auth.getToken()}`,
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+    };
+
+    return config;
+  },
+  (error) => {
+    Promise.reject(error);
+  }
+);
+
+instance.interceptors.response.use(
+  (response) => response,
+  (error) => {
+    // whatever you want to do with the error
+    if (error?.response?.status === 401) {
+      auth.clearAppStorage();
+      window.location.reload();
+    }
+
+    throw error;
+  }
+);
+
+const fetchClient = instance;
+
+export default fetchClient;
```

Comment:
I'd like to keep this interface as small and axios unspecific as possible. What do you think about exposing only the 4 HTTP methods?

```suggestion
export default {
  get: instance.get,
  post: instance.post,
  put: instance.put,
  delete: instance.delete
};
```

This would avoid some ambiguity we have at the moment, because axios allows to run `instance({ method: 'get' })` and `instance.get()`.

Reply:
ok I have modified the code and right now I am exposing these methods:
```
export default {
  getClient: instance.get,
  putClient: instance.put,
  postClient: instance.post,
  deleteClient: instance.delete,
};
```

do you think it is ok? what do you think about the names of the different methods I have used?

Reply:
I'm not much of a fan of these names. These are not clients, but function calls and I'd prefer plain HTTP verbs. Wdyt?